x/evm/handler.go

```diff
@@ -43,3 +46,9 @@ func Recover(err *error) {
 		}
 	}
 }
+
+func SetGasInContext(ctx sdk.Context, gasUsed uint64) {
```

Comment:
shouldn't you return the updated context here?

Reply:
we could also use the context directly from the keeper? Isn't the gas consumed twice here and in the `setGasMeterAndConsumeGas` function?

Reply:
thanks, your comment makes me understand some part of the logic that was bugging me at the beginning.

To sum up, ideally we should return and use the context from the keeper. That is why we are updating the gasMeter in the keeper. However there is two current issue to that

1- We cannot get easily the context from the keeper, but that can be fix with a cast (so its not really an issue)
2- Even if we get the context from the keeper, the cannot update the original context. The original context is pass to the handler but there is not way to update it afterward 
https://github.com/cosmos/cosmos-sdk/blob/master/baseapp/baseapp.go#L734

The solution that I am using now is updating the original context through `SetGasInContext`. This will directly update the object with the correct value (passed in reference) so no need to return anything.

Let me know if it satisfy your questions.

I think I should remove also the `setGasMeterAndConsumeGas` to update the gasMeter in the keeper because it is never used anywhere. Is it fine?